Include whether the managers in the swarm are autolocked as part of `docker info`

cyli opened the pull request:
Currently, the only way to tell if the cluster is autolocked is to attempt to unlock or get the unlock key.  The info is already available on the API for `docker info`, so also include it in the text output on the CLI.

cc @friism 

![image](https://i.pinimg.com/736x/b1/88/31/b18831f96720e907c4769168687d7fd1--cat-lovers-adorable-animals.jpg)

dnephin commented on cli/command/system/info_test.go:
```diff
@@ -0,0 +1,302 @@
+func newDockerCLI() (*command.DockerCli, *bytes.Buffer, *bytes.Buffer) {
+	var in, out, err bytes.Buffer
+	return command.NewDockerCli(ioutil.NopCloser(&in), &out, &err), &out, &err
+}
```
We have a fake that we use for testing at `internal/test/` `NewFakeCli()`

You can change `prettyPrintInfo()` to accept a `dockerCli command.Cli` interface instead of the struct

dnephin commented on cli/command/system/info_test.go:
```diff
@@ -0,0 +1,302 @@
+	require.NoError(t, prettyPrintInfo(cli, sampleInfoNoSwarm))
+	result, err := ioutil.ReadAll(stdout)
+	require.NoError(t, err)
+	require.Equal(t, []byte(sampleTextOutputNoSwarm), result)
```
I would recommend moving these sample texts to "golden files" (files in `./testdata` relative to this package directory) and using `gotestyourself/golden.Assert()` to compare them.  There are a bunch of example in this repo and the godoc is here: https://godoc.org/github.com/gotestyourself/gotestyourself/golden

This will make it easier to update and maintain the expected result, and it also uses `difflib` to give you a nice multiline diff when there are failures as long as you use the string version (not bytes). Comparing bytes gives you a diff of an array of uint8 which is impossible to debug.

dnephin commented on cli/command/system/info_test.go:
```diff
@@ -0,0 +1,302 @@
+	require.Equal(t, []byte(sampleTextOutputNoSwarm), result)
+	result, err = ioutil.ReadAll(stderr)
+	require.NoError(t, err)
+	require.Empty(t, result)
```
Using this fake this would be `assert.Equal(t, "", cli.ErrBuffer().String())`

dnephin commented on cli/command/system/info_test.go:
```diff
@@ -0,0 +1,302 @@
+	require.Empty(t, result)
+
+	// with swarm
+	cli, stdout, stderr = newDockerCLI()
```
Please make this a separate test function `TestPrettyPrintInfoWithSwarm()` which removes the need for this comment

dnephin commented on cli/command/system/info_test.go:
```diff
@@ -0,0 +1,302 @@
+	return command.NewDockerCli(ioutil.NopCloser(&in), &out, &err), &out, &err
+}
+
+func TestInfoNoWarnings(t *testing.T) {
```
Following our [testing guidelines](https://github.com/docker/cli/blob/master/TESTING.md#details) (which we just merged), this should be `TestPrettyPrintInfoNoSwarm()`

dnephin commented on cli/command/system/info_test.go:
```diff
@@ -0,0 +1,302 @@
+	infoWithSwarm := sampleInfoNoSwarm
+	infoWithSwarm.Swarm = sampleSwarmInfo
+	expected := []byte(
+		strings.Replace(sampleTextOutputNoSwarm, "Swarm: inactive", sampleTextOutputJustSwarm, 1))
```
I would just write out the exact file and compare using `golden` instead of doing a replace

dnephin commented on cli/command/system/info.go:
```diff
@@ -222,9 +223,11 @@ func prettyPrintInfo(dockerCli *command.DockerCli, info types.Info) error {
-		u := dockerCli.ConfigFile().AuthConfigs[info.IndexServerAddress].Username
-		if len(u) > 0 {
-			fmt.Fprintf(dockerCli.Out(), "Username: %v\n", u)
+		if dockerCli.ConfigFile() != nil {
```
If this is just because of the test, I think you can revert this. Using the `FakeCli` should give you a default config file I believe.

cyli commented:
Thanks for the review @dnephin!  I think I've addressed all the comments, and added a test for the warnings since the golden testdata files makes that way easier.
